static/js/app.js

```diff
@@ -676,6 +677,12 @@ $(function() {
                         text : ADD_DISCUSSION,
                         href : "#discussion-dialog",
                         callback : _.bind(this.showDiscussion, this)
+                    },
+                    {
+                        icon : "plus-sign",
+                        text : NEW_FEATURES,
+                        href : "#feature-dialog-template",
```

Comment:
You changed the `href` here to `feature-dialog-template`, but change the `id` of the template to `feature-dialog`...
I think the `id` of the modal should be `feature-dialog`, and the `div` containing it should be `feature-dialog-template` like it was before. Here the `href` should be `feature-dialog`, and then you don't need `this.modal.modal("show");` (see below).